sample/kotlin/src/main/kotlin/com/datadog/android/sample/data/DataRepository.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sample.data
+
+import com.datadog.android.sample.data.model.Post
+import retrofit2.Call
+import retrofit2.Response
+
+class DataRepository(private val remoteDataSource: RemoteDataSource) {
+
+    fun getPosts(callback: Callback<List<Post>>) {
+        val call = remoteDataSource.getSamplePosts()
+        call.enqueue(object : retrofit2.Callback<List<Post>> {
+            override fun onFailure(call: Call<List<Post>>, t: Throwable) {
+                callback.onFailure(Result.Failure(throwable = t))
+            }
+
+            override fun onResponse(call: Call<List<Post>>, response: Response<List<Post>>) {
+                val body = response.body()
+                if (response.isSuccessful && body != null) {
+                    callback.onSuccess(Result.Success(body))
+                } else {
+                    callback.onFailure(
+                        Result.Failure(
+                            response.errorBody()?.string() ?: "Request (${call.request().url()
+                                .url().path})" +
+                            " failed (${response.code()})"
+                        )
+                    )
+                }
+            }
+        })
+    }
```

Comment:
This method has too many nested blocks, maybe you could clean it up / split it a bit ?